tests/periph_wdt/main.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test for wdt Drivers
+ *
+ *              This test initializes the wdt counter to a preset value
+ *              and spins until a wdt reset is triggered
+ *
+ * @author      Francisco Molina <francois-xavier.molina@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "periph/wdt.h"
+#include "periph_cpu.h"
+#include "shell.h"
+#include "xtimer.h"
+
+int get_range(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+    printf("[wdt]: lower_bound: %d upper_bound: %d \n", NWDT_TIME_LOWER_LIMIT,
+           NWDT_TIME_UPPER_LIMIT);
+    return 0;
+}
+
+int setup_wdt(int argc, char **argv)
+{
+    if (argc < 3) {
+        printf("usage: %s <min_time[ms]> <max_time[ms]>\n", argv[0]);
+        return -1;
+    }
+
+    uint32_t min_time = atoi(argv[1]);
+    uint32_t max_time = atoi(argv[2]);
+
+    if (max_time > NWDT_TIME_UPPER_LIMIT || max_time < NWDT_TIME_LOWER_LIMIT) {
+        puts("[wdt]: invalid times, see \"range\"");
+        return -1;
+    }
+
+    wdt_setup_reboot(min_time, max_time);
+    puts("[wdt]: valid configuration");
+
+    return 0;
+}
+
+int start_wdt(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+    puts("[wdt]: starting wdt timer");
+    wdt_start();
+    return 0;
+}
+
+int start_loop_wdt(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+    printf("[wdt]: start time: %" PRIu32 " us\n", xtimer_now_usec());
+    wdt_start();
+    while (1) {
+        printf("[wdt]: reset time: %" PRIu32 " us\n", xtimer_now_usec());
```

Comment:
Prefixing with _[wdt]_ in the test application is not helpful to determine from the message comes from: the same prefix is used when enable debug in the peripheral implementation. Maybe just remove the prefix in the test application ? (and save some bytes on ROM).